tests/pytests/unit/modules/test_zypperpkg.py

```diff
@@ -254,3 +268,263 @@ def zypper_search_mock(name, *_args, **_kwargs):
+def test_dist_upgrade(package, pre_version, post_version, fromrepo_param):
+    with patch(
+        "salt.modules.zypperpkg.__zypper__.noraise.call"
+    ) as zypper_mock, patch.object(
+        zypper,
+        "list_pkgs",
+        MagicMock(side_effect=[{package: pre_version}, {package: post_version}]),
+    ):
+        expected_call = ["dist-upgrade", "--auto-agree-with-licenses"]
+
+        for repo in fromrepo_param:
+            expected_call.extend(["--from", repo])
+
+        result = zypper.upgrade(dist_upgrade=True, fromrepo=fromrepo_param)
+        zypper_mock.assert_any_call(*expected_call)
+        assert result == {package: {"old": pre_version, "new": post_version}}
+
+
+@pytest.mark.parametrize(
+    "package,pre_version,post_version,dup_avc,novendorchange_param,allowvendorchange_param,vendor_change",
+    [
+        # dup_avc = True, both params = default -> no vendor change
+        ("vim", "1.1", "1.2", True, True, False, False),
+        # dup_avc = True, allowvendorchange = True -> vendor change
+        (
+            "emacs",
+            "1.1",
+            "1.2",
+            True,
+            True,
+            True,
+            True,
+        ),
+        # dup_avc = True, novendorchange = False -> vendor change
+        ("joe", "1.1", "1.2", True, False, False, True),
+        # dup_avc = True, both params = toggled -> vendor change
+        ("kate", "1.1", "1.2", True, False, True, True),
+         # dup_avc = False -> no vendor change
+        (
+            "gedit",
+            "1.1",
+            "1.2",
+            False,
+            False,
+            True,
+            False
+        ),
+    ],
+)
+@patch.object(zypper, "refresh_db", MagicMock(return_value=True))
+def test_dist_upgrade_vendorchange(
+    package,
+    pre_version,
+    post_version,
+    dup_avc,
+    novendorchange_param,
+    allowvendorchange_param,
+    vendor_change
+):
+    cmd_run_mock = MagicMock(return_value={"retcode": 0, "stdout": None})
+    with patch.object(
+        zypper,
+        "list_pkgs",
+        MagicMock(side_effect=[{package: pre_version}, {package: post_version}]),
+    ), patch("salt.modules.zypperpkg.__zypper__.refresh_zypper_flags",), patch.dict(
+        zypper.__salt__, {"cmd.run_all": cmd_run_mock}
+    ):
+        expected_cmd = ["zypper", "--non-interactive", "--no-refresh", "dist-upgrade"]
+        # --allow-vendor-change is injected right after "dist-upgrade"
+        if vendor_change:
+            expected_cmd.append("--allow-vendor-change")
+        expected_cmd.append("--auto-agree-with-licenses")
+
+        zypper.__zypper__.dup_avc = dup_avc
+        zypper.upgrade(
+            dist_upgrade=True,
+            allowvendorchange=allowvendorchange_param,
+            novendorchange=novendorchange_param,
+        )
+        cmd_run_mock.assert_any_call(
+            expected_cmd, output_loglevel="trace", python_shell=False, env={}
+        )
```

Comment:
Same thing here. Do we need to upstream them?

Reply:
See above: yes either in https://github.com/saltstack/salt/pull/62033 or in https://github.com/saltstack/salt/pull/60421